server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/DomainQuotaRoutes.java

```diff
@@ -82,16 +88,26 @@ public DomainQuotaRoutes(DomainList domainList, DomainQuotaService domainQuotaSe
     public void define(Service service) {
         this.service = service;
 
-        defineGetQuotaCount();
-        defineDeleteQuotaCount();
-        defineUpdateQuotaCount();
+        if (isVirtualHostingSupported()) {
```

Comment:
it's not the way I expect things to happen.
I see two solutions : 
1. we decide to expose virtual hosting at guice level so we can inject objects based on its value
2. we always define these roots but we make sure lower level (here `data` module) throws the right exception.
The second one is much easier to discover IMO.

Reply:
ok